### DISCUSSION TITLE
en/tutorials/tuntap/

### DISCUSSION THREAD

**Discussion by giscus** [go-gost/gost#459] [General]
# en/tutorials/tuntap/

https://gost.run/en/tutorials/tuntap/

**Comment by PgRusher** [comment]
hi im so confused about this Tun/tap ok how can modify it with other protocols ? i mean how im able to use gost protocols with tun/tap ?

**Comment by PgRusher** [comment]
when im using Tun its create a Tun adapter and connected but how can i use it with other protocols ? to pass my data and when im using tap with your tutorials its nto working btw

**Comment by lukuichina** [comment]
请问route参数是不是不支持::/0？

**Comment by 0width** [comment]
如果A和B的网络都是192.168.3.0/24，A和B两个网络里的设备怎么通信

**Reply by PgRusher** [reply]
tun://:0/{serverURL}:3232?net=192.168.123.2/24&mtu=1350&name={adaptername}&keepalive=true&gw=192.168.123.1&log?level=error"
above one is in the client side and below is server side and with some setting you can make a Tun or Tap connection 
-L=tun://:8421?net=192.168.123.1/24

communicate like this easy


with this commands in debian linux servers :

sudo nano /etc/sysctl.conf
***************************
net.ipv4.ip_forward=1
=====================
iptables -t nat -A POSTROUTING -s 192.168.123.0/24 -o eth0 -j MASQUERADE
iptables -A FORWARD -i tun2 -o eth0 -j ACCEPT
iptables -A FORWARD -i eth0 -o tun2 -j ACCEPT
========================
sudo apt install iptables-persistent
***********************************
sudo netfilter-persistent save
**********************************
sudo iptables-save > /etc/iptables/rules.v4
sudo ip6tables-save > /etc/iptables/rules.v6
***********************************
sudo nano /etc/systemd/system/iptables-restore.service
**********************************
[Unit]
Description=Restore iptables rules
Before=network.target

[Service]
Type=oneshot
ExecStart=/sbin/iptables-restore /etc/iptables/rules.v4
ExecStart=/sbin/ip6tables-restore /etc/iptables/rules.v6
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
***********************************
sudo systemctl enable iptables-restore
**************************************
reboot


modify this ones as your server configs and also you need a wintun.dll along side this client codeto work and create a adapter in client side
